pkg/providers/gateway/translation/gateway_httproute.go

```diff
@@ -32,6 +32,77 @@ import (
 	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
 )
 
+func (t *translator) generatePluginsFromHTTPRouteFilter(filters []gatewayv1alpha2.HTTPRouteFilter) apisixv1.Plugins {
+	plugins := apisixv1.Plugins{}
+	for _, filter := range filters {
+		switch filter.Type {
+		case gatewayv1alpha2.HTTPRouteFilterRequestHeaderModifier:
+			t.fillPluginsFromHTTPRequestHeaderFilter(plugins, filter.RequestHeaderModifier)
+		case gatewayv1alpha2.HTTPRouteFilterRequestRedirect:
+			t.fillPluginsFromHTTPRequestRedirectFilter(plugins, filter.RequestRedirect)
+		case gatewayv1alpha2.HTTPRouteFilterRequestMirror:
+			// to do
+		}
+	}
+	return plugins
+}
+
+func (t *translator) fillPluginsFromHTTPRequestHeaderFilter(plugins apisixv1.Plugins, reqHeaderModifier *gatewayv1alpha2.HTTPRequestHeaderFilter) {
+	if reqHeaderModifier == nil {
+		return
+	}
+	headers := map[string]any{}
+	for _, header := range reqHeaderModifier.Add {
+		headers[string(header.Name)] = header.Value
```

Comment:
I think this behavior doesn't match the spec. `Add` is append, not overwrite. https://github.com/kubernetes-sigs/gateway-api/blob/release-0.4/apis/v1alpha2/httproute_types.go#L621

Reply:
I know, but I can't achieve it. What do you think should I do?